src/main/java/net/openhft/chronicle/bytes/internal/NativeBytesStore.java

```diff
@@ -839,8 +839,7 @@ public boolean sharedMemory() {
     @Override
     public long read(@NonNegative long offsetInRDI, byte[] bytes, @NonNegative int offset, @NonNegative int length) {
         requireNonNull(bytes);
-        // length is non-neg and an int
-        int len = (int) Math.min(length, readLimit() - offsetInRDI);
+        int len = Math.min(length, Maths.toUInt31(readLimit() - offsetInRDI));
```

Comment:
this reverts https://github.com/OpenHFT/Chronicle-Bytes/commit/b0324d677cdad0d74771c1f5fc871044e8be8add

We are better having this blow up if `readLimit() - offsetInRDI` is negative, rather than silently fail